### DISCUSSION TITLE
What "Apply code action" (`code_action`) do?

### DISCUSSION THREAD

**Discussion by NNBnh** [helix-editor/helix#1156] [Q&A] [answered]
From [keymap doc](https://docs.helix-editor.com/keymap.html):

> | Key | Description | Command |
> | - | - | - |
> | <kbd>a</kbd> | Apply code action | `code_action` |

But what does the code action really does, dose it complie and execute the code like [Coderun.kak](https://github.com/NNBnh/coderun.kak)?

Currently i couldn't figure out how to add a LSP to try it out, even following [the guide](https://docs.helix-editor.com/guides/adding_languages.html):

> ```sh
> git submodule add -f <repository> helix-syntax/languages/tree-sitter-<name>
> ```

I don't know where to run it...

```console
fatal: not a git repository (or any of the parent directories): .git
```

**Comment by sudormrfbin** [comment, marked as answer]
Code actions are provided by the LSP and should be supported by the particular language's LSP implementation (they are the little light bulbs that pop up beside lines in VSCode if that helps).

**Reply by 7ombie** [reply]
> they are the little light bulbs that pop up beside lines in VSCode if that helps

It did. Thanks.
